### PR TITLE
Mark webapi task failure as retry limit exceeded

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
fixed https://flyte-org.slack.com/archives/C05GQ16TQLF/p1692310075236329

propeller updates the `cache.state` to PhaseSystemFailure at [here](https://github.com/flyteorg/flyteplugins/blob/03f5b5c709277f49b04b65b591c469695defd8d4/go/tasks/pluginmachinery/internal/webapi/cache.go#L84). however, it sets the [action](https://github.com/flyteorg/flyteplugins/blob/03f5b5c709277f49b04b65b591c469695defd8d4/go/tasks/pluginmachinery/internal/webapi/cache.go#L94) to `cache.Unchanged` in `cache.ItemSyncResponse`, which won't update the item in the [LRU cache](https://github.com/flyteorg/flytestdlib/blob/dc81c279d511f271126443b2855843014727e34c/cache/auto_refresh.go#L288-L291).

Add an error message field to `cache.state`, and use it when we return `PhaseInfoFailure`. Therefore, we can see the error message on FlyteConsole if the propeller fails to get the task status.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
image

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: go/tasks/plugins/webapi/agent/plugin.go [hunk 1]
```diff
@@ -169,11 +171,17 @@ func (p Plugin) Status(ctx context.Context, taskCtx webapi.StatusContext) (phase
 	case admin.State_RETRYABLE_FAILURE:
 		return core.PhaseInfoRetryableFailure(pluginErrors.TaskFailedWithError, "failed to run the job", taskInfo), nil
 	case admin.State_SUCCEEDED:
+		var opReader io.OutputReader
 		if resource.Outputs != nil {
-			err := taskCtx.OutputWriter().Put(ctx, ioutils.NewInMemoryOutputReader(resource.Outputs, nil, nil))
-			if err != nil {
-				return core.PhaseInfoUndefined, err
-			}
+			logger.Infof(ctx, "Agent returned an output")
```
Comment (pingsutw) [inline go/tasks/plugins/webapi/agent/plugin.go:176]: Note: remove it once this [pr](https://github.com/flyteorg/flyteplugins/pull/391) is merged.

**Comment by honnix** [comment]
Thank you for address this. Does this also make `CreateTask` failure handled correctly? More concretely [here](https://github.com/flyteorg/flyteplugins/blob/552f145c4632944a23e7e2d9fa2b2f065c4b14b9/go/tasks/pluginmachinery/internal/webapi/launcher.go#L19). Will that error be handled correctly?

**Comment by pingsutw** [comment]
> Does this also make CreateTask failure handled correctly? More concretely [here](https://github.com/flyteorg/flyteplugins/blob/552f145c4632944a23e7e2d9fa2b2f065c4b14b9/go/tasks/pluginmachinery/internal/webapi/launcher.go#L19). Will that error be handled correctly?

That's another issue, I'll fix it in a separate PR.

**Review by hamersaw — APPROVED** [review]
